spec/lib/new_post_manager_spec.rb

```diff
@@ -107,11 +107,12 @@
 
   describe "default handler" do
     let(:manager) { NewPostManager.new(user, raw: "this is new post content", topic_id: topic.id) }
+    before { Group.refresh_automatic_groups! }
```

Comment:
When is "automatic groups refreshing" necessary? Is there any chance this should happen after each group is fabricated? What impact does it have?

Reply:
It's used more for fabricating users because they aren't automatically added to the appropriate trust_level_x groups. From my experience we should favor `fab!(:user) { Fabricate(:user, refresh_auto_groups: true) }` over `Group.refresh_automatic_groups!` and only use the latter when we can't use the former. But Yes in this specific case we can actually remove the use of `Group.refresh_automatic_groups!`.

Reply:
Yes the `refresh_auto_groups: true` which I added recently is better for sure, or at least more targeted.